pommerman/network/ion_client/__init__.py

```diff
@@ -0,0 +1,219 @@
+from .network import Network
+import signal
+
+
+def _exit_handler(_s=None, _h=None):
+    """Arguments:  
+    * _s: Unused argument  
+    * _h: Unused argument  
+    Description: Handle exiting the application"""
+    ui.info(ui.yellow, "Exiting..")
+    exit(0)
+
+
+def init():
+    """Description: Starts up the application normally by asking the user about
+    the server they want to connect to"""
+    signal.signal(signal.SIGINT, _exit_handler)
+    if ui.ask_yes_no(Strings.server_prompt.value):
+        domain = ui.ask_string(Strings.server_ip.value)
+        if domain is None:
+            ui.fatal(Strings.error_invalid_domain.value)
+    else:
+        domain = "play.pommerman.com:5050"
+    ui.info(
+        Strings.server_connecting_p1.value, ui.yellow,
+        Strings.server_connecting_p2.value, ui.reset,
+        Strings.server_connecting_p3.value
+    )
+    network = Network(domain)
+    try:
+        status = network.check()
+    except Exception as e:
+        ui.fatal(e)
+    ui.info(
+        Strings.server_connected.value, ui.yellow, Strings.server_players.value,
+        str(status[0]) + ",", Strings.server_matches.value, status[1]
+    )
+    intent(network)
+
+
+def intent(network):
+    """Arguments:   
+    * network: An `network`(pommerman.network.ion_client.network) object  
+    Description: This creates a prompt for the user where they can choose to:  
+    * Play a match  
+    * Create/Join a room  
+    * Replay a match  
+    * Exit the application
+    """
+    i = ui.ask_choice(
+        Strings.intent.value, [
+            Strings.intent_match.value, Strings.intent_room.value,
+            Strings.intent_replay.value, Strings.intent_exit.value
+        ]
+    )
+    if i == Strings.intent_match.value:
+        match(network)
+    elif i == Strings.intent_room.value:
+        match(network, True)
+    elif i == Strings.intent_replay.value:
+        replay(network)
+    elif i == Strings.intent_exit.value:
+        exit(0)
+
+
+def match(
+    network, room=False, import_name=False, agent_class=False, ui_en=True
+):
+    """Arguments:  
+    * network: An `network`(pommerman.network.ion_client.network) object  
+    * room: The room to be created/joined. If False, the public room will be \
+joined  
+    * import_name: The module which contains your agent  
+    * agent_class: The class of the agent within the module  
+    * ui_en: If the UI is enabled or disabled (This also controls if exception \
+are raised or not)  
+    Description: This facilitates playing a match"""
+    # Note: Imports are local as they aren't required anywhere else
+    from os import getcwd
+    from sys import path
+    from importlib import import_module
+    from gym.spaces import Discrete
+    from pommerman.constants import Item
+    if room:
+        room = str(ui.ask_string(Strings.room_code.value))
+    # Info: The Current Working Directory has to be added to path to import modules from it
+    path.append(getcwd())
```

Comment:
From here down a bit (to 118) is mostly just `safely get the agent`. Plucking that out into its own private function will help w clarity.

Reply:
A 7 line function, that'll just create more confusion for such a small part of the code.